### ISSUE TITLE
Cannot mount VMware tools ISO

### ISSUE THREAD

**Issue by sdorsett** [mitchellh/packer#1795]
Using packer 7.5 with vmware-iso builder on Centos 6.5 installing ESXi 5.5 on a remote ESXi host. VNC shows the virtual machine properly installs, but the following errors show at the end of the install process:

```
014/12/26 17:52:56 ui: ==> esxi55: Compacting the disk image
2014/12/26 17:52:56 packer-builder-vmware-iso: 2014/12/26 17:52:56 remote command exited with '0': vim-cmd vmsvc/reload 22
==> esxi55: Compacting the disk image
2014/12/26 17:52:56 packer-builder-vmware-iso: 2014/12/26 17:52:56 opening new ssh session
2014/12/26 17:52:56 packer-builder-vmware-iso: 2014/12/26 17:52:56 starting remote command: vim-cmd vmsvc/power.getstate 22
2014/12/26 17:52:56 ui: ==> esxi55: Unregistering virtual machine...
2014/12/26 17:52:56 packer-builder-vmware-iso: 2014/12/26 17:52:56 remote command exited with '0': vim-cmd vmsvc/power.getstate 22
==> esxi55: Unregistering virtual machine...
2014/12/26 17:52:56 packer-builder-vmware-iso: 2014/12/26 17:52:56 opening new ssh session
2014/12/26 17:52:56 packer-builder-vmware-iso: 2014/12/26 17:52:56 starting remote command: vim-cmd vmsvc/unregister 22
2014/12/26 17:52:57 ui error: Build 'esxi55' errored: Couldn't mount VMware tools ISO. Please check the 'guest_os_type' in your template.json.
2014/12/26 17:52:57 Builds completed. Waiting on interrupt barrier...
2014/12/26 17:52:57 machine readable: error-count []string{"1"}
2014/12/26 17:52:57 ui error:
==> Some builds didn't complete successfully and had errors:
2014/12/26 17:52:57 machine readable: esxi55,error []string{"Couldn't mount VMware tools ISO. Please check the 'guest_os_type' in your template.json."}
2014/12/26 17:52:57 ui error: --> esxi55: Couldn't mount VMware tools ISO. Please check the 'guest_os_type' in your template.json.
2014/12/26 17:52:57 ui:
==> Builds finished but no artifacts were created.
2014/12/26 17:52:57 waiting for all plugin processes to complete...
2014/12/26 17:52:57 /root/packer/7.5/packer-post-processor-vagrant-vmware-ovf: plugin process exited
2014/12/26 17:52:57 /root/packer/7.5/packer-provisioner-file: plugin process exited
2014/12/26 17:52:57 /root/packer/7.5/packer-provisioner-shell: plugin process exited
2014/12/26 17:52:57 /root/packer/7.5/packer-provisioner-shell: plugin process exited
2014/12/26 17:52:57 packer-builder-vmware-iso: 2014/12/26 17:52:57 remote command exited with '0': vim-cmd vmsvc/unregister 22
Build 'esxi55' errored: Couldn't mount VMware tools ISO. Please check the 'guest_os_type' in your template.json.

==> Some builds didn't complete successfully and had errors:
--> esxi55: Couldn't mount VMware tools ISO. Please check the 'guest_os_type' in your template.json.

==> Builds finished but no artifacts were created.
2014/12/26 17:52:57 /root/packer/7.5/packer-builder-vmware-iso: plugin process exited
```

The packer template I'm using specifies vmkernel5 for the guest_os_type, which doesn't have a VM tools ISO for tools install, but a shell provisioner handles installing vmware tools at the end of the install. The following is the packer template:

```
[root@packer packer-templates]# cat templates/esxi.json
{
  "variables": {
    "version": "1.0"
  },
  "builders": [
    {
      "name": "esxi55",
      "vm_name": "esxi55",
      "vmdk_name": "esxi55-disk0",
      "type": "vmware-iso",
      "headless": true,
      "disk_size": 4096,
      "guest_os_type": "vmkernel5",
      "iso_url": "./isos/VMware-VMvisor-Installer-5.5.0-1331820.x86_64.iso",
      "iso_checksum": "ef599dc7e647177027684c0eee346ccdbc8704f2",
      "iso_checksum_type": "sha1",
      "remote_host": "192.168.1.5",
      "remote_datastore": "local-sata",
      "remote_username": "root",
      "remote_password": "*******",
      "remote_type": "esx5",
      "ssh_username": "root",
      "ssh_password": "vagrant",
      "ssh_wait_timeout": "60m",
      "shutdown_command": "esxcli system maintenanceMode set -e true -t 0 ; esxcli system shutdown poweroff -d 10 -r 'Packer Shutdown' ; esxcli system maintenanceMode set -e false -t 0",
      "http_directory": ".",
      "boot_wait": "5s",
      "vmx_data": {
        "memsize": "4096",
        "numvcpus": "2",
        "vhv.enable": "TRUE",
        "ethernet0.virtualDev": "e1000",
        "ethernet0.networkName": "vlan2",
        "ethernet0.present": "TRUE"

      },
      "boot_command": [
        "<enter><wait>O<wait> ks=http://{{ .HTTPIP }}:{{ .HTTPPort }}/scripts/esxi-5-kickstart.cfg<enter>"
      ]
    }
  ],
  "provisioners": [
    {
      "type": "file",
      "source": "puppet/modules/vagrantbaseconfig/files/vagrant.pub",
      "destination": "/etc/ssh/keys-root/authorized_keys"
    },
    {
      "type": "shell",
      "script": "scripts/esxi-vmware-tools_install.sh"
    },
    {
      "type": "shell",
      "script": "scripts/esxi-cloning_configuration.sh"
    }
  ],
  "post-processors": [
   {
     "type": "vagrant-vmware-ovf",
     "compression_level": 9,
     "output": "{{.BuildName}}-{{.Provider}}-{{user `version`}}.box"
   }
  ]
}
```

This template, without the remote_ options, succeeds with packer and fusion on OS X.

**Comment by michaeltlombardi** [comment]
@sdorsett Is this problem ongoing or has it been fixed by an update to packer/post-processors since then?

**Comment by rickard-von-essen** [comment]
Closing since the original issue is solved.

**Comment by berney** [comment]
I'm hitting the same issue.

Using packer 0.10.2, vmware_guest_os_type: darwin12-64, vmware-iso building on ESXi 5.5 target. Packer tries to install VMware Tools using vim-cmd, but this fails as the OSX Installation CD is inserted. After the provisioning steps complete, packer fails  the build because the vim-cmd to install VMware Tools failed.

Packer should try to eject the CD or allow for provisioing scripts to install VMware Tools. Really it should be doing a functionality test to determine that VMware Tools is working, and it should allow the user to override, so that a VM can be built that doesn't have VMware Tools installed.

**Comment by rickard-von-essen** [comment]
@berney Could you please open a new issue.
